octobot/community/authentication.py

```diff
@@ -357,6 +376,61 @@ async def get_subscribed_profile_urls(self):
             for profile_data in subscribed_profiles["data"]
         ]
 
+    async def update_trades(self, trades: list):
+        """
+        Updates authenticated account trades
+        """
+        if not self.is_logged_in():
+            return
+        try:
+            formatted_trades = [
+                {
+                    "date": f"{datetime.datetime.fromtimestamp(trade.executed_time).isoformat('T')}Z",
+                    "exchange": trade.exchange_manager.exchange_name,
+                    "price": str(trade.executed_price),
+                    "quantity": str(trade.executed_quantity),
+                    "symbol": trade.symbol,
+                    "type": trade.trade_type.value,
+                }
+                for trade in trades
+            ]
+            await self._update_bot_trades(formatted_trades)
+        except Exception as err:
+            self.logger.exception(err, True, f"Error when updating community trades {err}")
+
+    async def update_portfolio(self, current_value: dict, initial_value: dict,
+                               unit: str, content: dict, history: dict, price_by_asset: dict):
+        """
+        Updates authenticated account portfolio
+        """
+        if not self.is_logged_in():
+            return
+        try:
+            ref_market_current_value = current_value[unit]
+            ref_market_initial_value = initial_value[unit]
+            formatted_content = [
+                {
+                    "asset": key,
+                    "quantity": str(quantity[commons_constants.PORTFOLIO_TOTAL]),
+                    "value": str(quantity[commons_constants.PORTFOLIO_TOTAL] * float(price_by_asset.get(key, 0))),
+                }
+                for key, quantity in content.items()
+            ]
+            formatted_history = [
+                {
+                    "date": f"{datetime.datetime.fromtimestamp(key).isoformat('T')}Z",
```

Comment:
Should we create a method that converts a timestamp to a graphql datetime? to prevent code duplication